src/server/modules/mongodb-listener/listener.service.ts

```diff
@@ -45,20 +46,22 @@ export class ListenerService extends EventEmitter {
             serverSelectionTimeoutMS: 1000,
           },
         );
+
+        // Listen for the 'close' event on the MongoClient instance
+        this.client.on('close', () => {
+          this.logger.warn('MongoDB connection lost');
+          this.emit('end');
+          this.connectToMongoDB();
+        });
+
         this.logger.success('Successfully connected to MongoDB instance');
+        this.emit('connection');
         return;
       } catch (error) {
-        retries--;
-        if (!retries) {
-          this.logger.error(
-            `Failed to connect to MongoDB instance: ${error.message}`,
-          );
-        } else {
-          this.logger.warn(
-            'Failed to connect: retrying to connect to MongoDB in 1 second...',
-          );
-          await new Promise((resolve) => setTimeout(resolve, 1000));
-        }
+        this.logger.warn(
+          `Failed to connect: retrying to connect to MongoDB in 1 second... Error: ${error.message}`,
+        );
+        await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
This should be tested in integration test listener.test.ts. Why is it not the case ?

Reply:
This part was not backed by an integration test. I did it.